include/libdnf/utils/exception.hpp

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_UTILS_EXCEPTION_HPP
+#define LIBDNF_UTILS_EXCEPTION_HPP
+
+
+#include <stdexcept>
+#include <system_error>
+
+namespace libdnf {
+
+class Exception : public std::runtime_error {
+public:
+    using runtime_error::runtime_error;
+    virtual const char * get_domain_name() const noexcept { return "libdnf"; }
+    virtual const char * get_name() const noexcept { return "Exception"; }
+    virtual const char * get_description() const noexcept { return ""; }
+};
```

Comment:
Shell we extend all errors by error code value?

Reply:
Example:
An `Exception` with error code `bad_value` can be replaced by inherited specialized exception `BadValue`.

I thought about adding error code for easy transform to other languages eg C. The same approach as get_name() -> Each exception class will have unique `id` (`error code`). But how create well defined unique `id`? The string `get_domain_name() + get_name()` is unique to each exception now.

Note:
There is `error_code` in my SystemError exception class. Because system error codes are defined and I think that is no need to create specialized exception for each system error code.
I have some concept but it can be modified.